### DISCUSSION TITLE
Symlinks in src/pages, multiple pages source directories

### DISCUSSION THREAD

**Discussion by iserdmi** [vikejs/vike#1674] [Discussion]
### Description

Hello! I want to place my pages files not only inside "./pages" or "./src/pages". I also want to be possible store it inside "../../another-package/src/etc/my-page". I was expected if I do `ln -s ../../another-package/src/etc ./src/pages/another-package-symlink` that will work out of the box. So I want to have many places to store my pages. How can I do it now? Is it possible? Thanks a lot!

**Comment by iserdmi** [comment]
It will be perfect to have setting like `pagesDirsGlobs`. For example looks like by default now it is `["./pages/**/*", "./pages/**/*"]`. In my case I want make it `["./pages/**/*", "./pages/**/*", "../../another-package/src/**/*Page"]`

**Reply by brillout** [reply]
You can already do that with https://vike.dev/filesystemRoutingRoot.

**Reply by iserdmi** [reply]
@brillout Thanks for link! As I understand this method allows only to change one pages fir to another. 
1. What if I want multiple roots like "./pages" as default, and also "../../another-package/src/pages"
2. What if my "../../another-package/src" does not contain pages dir. It contains a lot of dirs with a lot of client and server files, but some dirs called "BlaBlaBlaPage", and I want vike just watch to that dirs which suitable to glob "../../another-package/src/**/*Page"
Help me please. Thanks a lot! Vike is cool! 🙂

**Reply by brillout** [reply]
I believe that cdb7bef475b7f0529cdea30335d3099f09d86334 and most notably https://vike.dev/filesystem-routing#crawl answers your question?

**Reply by iserdmi** [reply]
@brillout Sorry, maybe I am missing something, but I still cannot make it work using ln -s. I have created a minimal representation with two packages in a monorepo. It can be a good example for others. However, the linked pages from the "shared" package still do not work inside the "webapp" (vike) package. I would really appreciate it if you could help me: https://github.com/iserdmi/vike-link-example/tree/master (It has README.md)

**Reply by brillout** [reply]
The issue is that the content of symlink directories isn't listed by `$ git ls-files` (which Vike uses).

Is there a Git command to list the the content of symlink directories?

**Reply by iserdmi** [reply]
@brillout I was trying to find a Git command that would also list files inside a symlink directory, but I couldn't find one. This is logical because the files inside a symlink are not necessarily tracked by the Git repository. At the same time, it's important to note that the symlink directory itself should probably be covered by .gitignore because it is absolute in relation to the host's file system. Such a symlink directory should be created automatically during project initialization, for example, through a "prepare" script. So Git won't see it at all.

**Reply by brillout** [reply]
That makes sense in principle, but in practice `git ls-files` is actually able to list non-tracked files (which is required for being able to use `git ls-files` for crawling).

Would you be up for a PR? The crawling code is self isolated and is thus relatively easy to change. The command is created here: https://github.com/vikejs/vike/blob/30a09690a670f66d921aaf3fdcf30bd94e7e06e5/vike/node/plugin/plugins/importUserCode/v1-design/getVikeConfig/crawlPlusFiles.ts#L111

One possibility is to add the `-s` flag (`git ls-files -s`) to see if there is a symlink, then fallback to `fast-glob` if there is one. But I'd rather stick to Git if possible.

**Reply by iserdmi** [reply]
@brillout Ok, I will try, thanks! May be it will be not best realisation, but I'll try my best.

**Reply by iserdmi** [reply]
@brillout Done. Looks nice, I think 🙂 Please check it: https://github.com/vikejs/vike/pull/1688

**Comment by brillout** [comment]
AFAICT `ln -s` should work just fine.

**Comment by iserdmi** [comment]
I have package "general" with my business code for backend and fronted sorted by feature inside dirs. inside some dirs I want to have dirs called "BlaBlaBlaPage" and I want my package "webapp" with vike itself can see that dirs and ignore other dirs which not has "*Page" suffix. Also I am not using file system routing. I use route functions always. Als some pages I want still hold inside my "webapp" package inside my monorepo.
<img width="383" alt="image" src="https://github.com/vikejs/vike/assets/4304927/2bb2a944-a128-4bd8-a061-4ee7da7f79cd">
<img width="378" alt="image" src="https://github.com/vikejs/vike/assets/4304927/7ba28623-fb45-450c-a2d8-38369b2bf4cc">
As I understand with https://vike.dev/filesystemRoutingRoot I can switch one dir to another. I need multiple dirs as I understand. Also I have tried make `ln -s '/Users/iserdmi/Dropbox/Code/projects/estatelink/packages/general/src/property/pages/Test1Page' ./Test1Page` and this page still not available (page not found). Help me please, give me some advice. Thanks!

**Reply by brillout** [reply]
I'm not sure what you mean here, but see my sibling comment.

**Comment by iserdmi** [comment]
I have created a minimal representation with two packages in a monorepo. Where one package uses vike. And pages stores in another package. It can be a good example for others. However, the linked pages from the "shared" package still do not work inside the "webapp" (vike) package. I would really appreciate it if somebody could help me: https://github.com/iserdmi/vike-link-example/tree/master (It has README.md)

**Reply by brillout** [reply]
I could indeed reproduce. I think I can fix it.
